### PR TITLE
"Implement FixedPoint trait" and  "Enable fixed point u128"

### PR DESCRIPTION
**Pull request by aliXsed**
Bring in https://github.com/paritytech/substrate/pull/5877 and https://github.com/paritytech/substrate/pull/6214 as preparational steps for #141:

* Resolve the merge conflicts
* Remove saturating_pow for now to avoid further extension of the change scope 

Co-authored-by: Gavin Wood <gavin@parity.io>
Co-authored-by: Shawn Tabrizi <shawntabrizi@gmail.com>

### CODE REVIEW

File: bin/node/executor/tests/fees.rs [hunk 1]
```diff
@@ -202,7 +202,7 @@ fn transaction_fee_is_correct_ultimate() {
 fn block_weight_capacity_report() {
 	// Just report how many transfer calls you could fit into a block. The number should at least
 	// be a few hundred (250 at the time of writing but can change over time). Runs until panic.
-	use node_primitives::Index;
+	use node_primitives::{Index, Hash};
```
Comment (jordy25519) [inline bin/node/executor/tests/fees.rs:205]: (nit) wondering why this change?

**Review by jordy25519 — APPROVED** [review]
lgtm!

**Review by KarishmaBothara — APPROVED** [review]
